// infermedicaapilib/lib/Models/DiagnosisResponse.js

/**
 * InfermedicaAPILib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io ) on 02/11/2017
 */
var BaseModel = require("./BaseModel");
/**
 * Creates an instance of DiagnosisResponse
 *
 * @constructor
 */
DiagnosisResponse = function (obj) {
    if(!obj) {
        this.conditions = null;     
        this.extras = null;     
        this.question = null;     
    } else {
        this.conditions = (obj.conditions !== undefined && obj.conditions !== null)? obj.conditions.map(function(model){
            return new ConditionProbability(model);
        }) : null;
        this.extras = (obj.extras !== undefined && obj.extras !== null)? obj.extras : null;
        this.question = (obj.question !== undefined && obj.question !== null)? new Question(obj.question) : null;
    }
};

DiagnosisResponse.prototype = new BaseModel();
DiagnosisResponse.prototype.constructor = DiagnosisResponse;

/**
 * TODO: Write general description for this method
 *
 * @return {array|null}
 */
DiagnosisResponse.prototype.getConditions = function() {
    return this.conditions;
};

/**
 * Setter for Conditions
 * 
 * @param {array|null} value 
 */
DiagnosisResponse.prototype.setConditions = function(value) {
    this.conditions = value;
};

/**
 * TODO: Write general description for this method
 *
 * @return {object|null}
 */
DiagnosisResponse.prototype.getExtras = function() {
    return this.extras;
};

/**
 * Setter for Extras
 * 
 * @param {object|null} value 
 */
DiagnosisResponse.prototype.setExtras = function(value) {
    this.extras = value;
};

/**
 * TODO: Write general description for this method
 *
 * @return {Question|null}
 */
DiagnosisResponse.prototype.getQuestion = function() {
    return this.question;
};

/**
 * Setter for Question
 * 
 * @param {Question|null} value 
 */
DiagnosisResponse.prototype.setQuestion = function(value) {
    this.question = value;
};


module.exports = DiagnosisResponse;
