// infermedicaapilib/lib/Models/Evidence.js

/**
 * InfermedicaAPILib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io ) on 02/11/2017
 */
var BaseModel = require("./BaseModel");
/**
 * Creates an instance of Evidence
 *
 * @constructor
 */
Evidence = function (obj) {
    if(!obj) {
        this.choiceId = null;     
        this.id = null;     
        this.observedAt = null;     
        //Append to variable dictionary
        this._variableDict['choiceId'] = 'choice_id';
        this._variableDict['observedAt'] = 'observed_at';
    } else {
        this.choiceId = (obj.choice_id !== undefined && obj.choice_id !== null)? obj.choice_id : null;
        this.id = (obj.id !== undefined && obj.id !== null)? obj.id : null;
        this.observedAt = (obj.observed_at !== undefined && obj.observed_at !== null)? obj.observed_at : null;
        //Append to variable dictionary
        this._variableDict['choiceId'] = 'choice_id';
        this._variableDict['observedAt'] = 'observed_at';
    }
};

Evidence.prototype = new BaseModel();
Evidence.prototype.constructor = Evidence;

/**
 * TODO: Write general description for this method
 *
 * @return {ChoiceIdEnum}
 */
Evidence.prototype.getChoiceId = function() {
    return this.choiceId;
};

/**
 * Setter for ChoiceId
 * 
 * @param {ChoiceIdEnum} value 
 */
Evidence.prototype.setChoiceId = function(value) {
    this.choiceId = value;
};

/**
 * id of observation, risk factor or condition
 *
 * @return {string}
 */
Evidence.prototype.getId = function() {
    return this.id;
};

/**
 * Setter for Id
 * 
 * @param {string} value 
 */
Evidence.prototype.setId = function(value) {
    this.id = value;
};

/**
 * time when evidence was observed in ISO 8601 format
 *
 * @return {string|null}
 */
Evidence.prototype.getObservedAt = function() {
    return this.observedAt;
};

/**
 * Setter for ObservedAt
 * 
 * @param {string|null} value 
 */
Evidence.prototype.setObservedAt = function(value) {
    this.observedAt = value;
};


module.exports = Evidence;
