// infermedicaapilib/lib/Models/Question.js

/**
 * InfermedicaAPILib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io ) on 02/11/2017
 */
var BaseModel = require("./BaseModel");
/**
 * Creates an instance of Question
 *
 * @constructor
 */
Question = function (obj) {
    if(!obj) {
        this.items = null;     
        this.text = null;     
        this.type = null;     
    } else {
        this.items = (obj.items !== undefined && obj.items !== null)? obj.items.map(function(model){
            return new QuestionItem(model);
        }) : null;
        this.text = (obj.text !== undefined && obj.text !== null)? obj.text : null;
        this.type = (obj.type !== undefined && obj.type !== null)? obj.type : null;
    }
};

Question.prototype = new BaseModel();
Question.prototype.constructor = Question;

/**
 * one or more observations to evaluate
 *
 * @return {array}
 */
Question.prototype.getItems = function() {
    return this.items;
};

/**
 * Setter for Items
 * 
 * @param {array} value 
 */
Question.prototype.setItems = function(value) {
    this.items = value;
};

/**
 * question text
 *
 * @return {string}
 */
Question.prototype.getText = function() {
    return this.text;
};

/**
 * Setter for Text
 * 
 * @param {string} value 
 */
Question.prototype.setText = function(value) {
    this.text = value;
};

/**
 * TODO: Write general description for this method
 *
 * @return {TypeEnum|null}
 */
Question.prototype.getType = function() {
    return this.type;
};

/**
 * Setter for Type
 * 
 * @param {TypeEnum|null} value 
 */
Question.prototype.setType = function(value) {
    this.type = value;
};


module.exports = Question;
