// infermedicaapilib/lib/Models/QuestionItem.js

/**
 * InfermedicaAPILib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io ) on 02/11/2017
 */
var BaseModel = require("./BaseModel");
/**
 * Creates an instance of QuestionItem
 *
 * @constructor
 */
QuestionItem = function (obj) {
    if(!obj) {
        this.choices = null;     
        this.id = null;     
        this.name = null;     
    } else {
        this.choices = (obj.choices !== undefined && obj.choices !== null)? obj.choices.map(function(model){
            return new Choice(model);
        }) : null;
        this.id = (obj.id !== undefined && obj.id !== null)? obj.id : null;
        this.name = (obj.name !== undefined && obj.name !== null)? obj.name : null;
    }
};

QuestionItem.prototype = new BaseModel();
QuestionItem.prototype.constructor = QuestionItem;

/**
 * list of available answer choices
 *
 * @return {array}
 */
QuestionItem.prototype.getChoices = function() {
    return this.choices;
};

/**
 * Setter for Choices
 * 
 * @param {array} value 
 */
QuestionItem.prototype.setChoices = function(value) {
    this.choices = value;
};

/**
 * observation id
 *
 * @return {string}
 */
QuestionItem.prototype.getId = function() {
    return this.id;
};

/**
 * Setter for Id
 * 
 * @param {string} value 
 */
QuestionItem.prototype.setId = function(value) {
    this.id = value;
};

/**
 * name or alias of observation
 *
 * @return {string}
 */
QuestionItem.prototype.getName = function() {
    return this.name;
};

/**
 * Setter for Name
 * 
 * @param {string} value 
 */
QuestionItem.prototype.setName = function(value) {
    this.name = value;
};


module.exports = QuestionItem;
