// infermedicaapilib/lib/Models/RiskFactor.js

/**
 * InfermedicaAPILib
 *
 * This file was automatically generated by APIMATIC v2.0 ( https://apimatic.io ) on 02/11/2017
 */
var BaseModel = require("./BaseModel");
/**
 * Creates an instance of RiskFactor
 *
 * @constructor
 */
RiskFactor = function (obj) {
    if(!obj) {
        this.id = null;     
        this.name = null;     
    } else {
        this.id = (obj.id !== undefined && obj.id !== null)? obj.id : null;
        this.name = (obj.name !== undefined && obj.name !== null)? obj.name : null;
    }
};

RiskFactor.prototype = new BaseModel();
RiskFactor.prototype.constructor = RiskFactor;

/**
 * TODO: Write general description for this method
 *
 * @return {string}
 */
RiskFactor.prototype.getId = function() {
    return this.id;
};

/**
 * Setter for Id
 * 
 * @param {string} value 
 */
RiskFactor.prototype.setId = function(value) {
    this.id = value;
};

/**
 * TODO: Write general description for this method
 *
 * @return {string}
 */
RiskFactor.prototype.getName = function() {
    return this.name;
};

/**
 * Setter for Name
 * 
 * @param {string} value 
 */
RiskFactor.prototype.setName = function(value) {
    this.name = value;
};


module.exports = RiskFactor;
